starboard/nplb/maximum_player_configuration_explorer.cc

```diff
@@ -0,0 +1,242 @@
+    auto& it = queue.front();
+    auto& config = it.first;
+    auto& index = it.second;
+    while (index < resources_) {
+      if (config[index] != instances_) {
+        ++config[index];
+        if (advancable.find(config) != advancable.end()) {
+          --config[index];
+        } else {
+          break;
+        }
+      }
+      ++index;
+    }
+
+    if (index != resources_) {
+      if (test_functor_(config)) {
+        queue.push_front(std::make_pair(config, index));
+        advancable[config] = true;
+      } else {
+        advancable[config] = false;
+      }
+      --config[index];
+      ++index;
+    } else {
+      bool can_advance = false;
+      for (int i = 0; i < config.size(); ++i) {
+        if (config[i] >= instances_) {
+          continue;
+        } else {
+          ++config[i];
+          if (advancable[config]) {
+            can_advance = true;
+            break;
+          }
+          --config[i];
+        }
+      }
+      if (!can_advance) {
+        max_test_config_set_.insert(config);
+      }
+      queue.pop_front();
+    }
+  }
+}
+
+MaximumPlayerConfigurationExplorer::MaximumPlayerConfigurationExplorer(
+    const std::set<PlayerConfig>& player_configs,
+    testing::FakeGraphicsContextProvider* fake_graphics_context_provider)
+    : fake_graphics_context_provider_(fake_graphics_context_provider) {
+  SB_DCHECK(!player_configs.empty());
+
+  player_configs_.assign(player_configs.begin(), player_configs.end());
+  Init();
+}
+
+MaximumPlayerConfigurationExplorer::MaximumPlayerConfigurationExplorer(
+    const std::set<PlayerConfig>& player_configs,
+    const std::set<std::vector<int>>& pseudo_max_element_set,
+    testing::FakeGraphicsContextProvider* fake_graphics_context_provider)
+    : fake_graphics_context_provider_(fake_graphics_context_provider) {
+  SB_DCHECK(!player_configs.empty());
+  SB_DCHECK(!pseudo_max_element_set.empty());
+
+  player_configs_.assign(player_configs.begin(), player_configs.end());
+  pseudo_max_element_set_.assign(pseudo_max_element_set.begin(),
+                                 pseudo_max_element_set.end());
+  for (int i = 0; i < pseudo_max_element_set_.size(); ++i) {
+    for (int j = i + 1; j < pseudo_max_element_set_.size(); ++j) {
+      SB_DCHECK(!PosetInSameChain(pseudo_max_element_set_[i],
+                                  pseudo_max_element_set_[j]));
+    }
+  }
+
+  Init();
+}
+
+MaximumPlayerConfigurationExplorer::~MaximumPlayerConfigurationExplorer() {
+  for (const auto& players_of_config : players_) {
+    for (auto player : players_of_config) {
+      SbPlayerDestroy(player);
+    }
+  }
+}
+
+void MaximumPlayerConfigurationExplorer::Init() {
+  players_.resize(player_configs_.size());
+  current_config_.resize(player_configs_.size(), 0);
+
+  std::function<bool(const std::vector<int>&)> test_functor = std::bind(
+      &MaximumPlayerConfigurationExplorer::Create, this, std::placeholders::_1);
+  if (!pseudo_max_element_set_.empty()) {
```

Comment:
Use `if (pseudo_max_element_set_.empty()) {` and reverse the if/else bodies.

Reply:
done and pseudo_max_element_set_ is removed; replaced by |terminate_functor_|